### ISSUE TITLE
Combobox: unintuitive behaviour on re-selection

### ISSUE THREAD

**Issue by d-m-s** [vmware-clarity/ng-clarity#386]
## References
@bbogdanov re-raised the issue.
See these first: 
- https://github.com/vmware/clarity/issues/6469#issue-1052060434
- https://github.com/vmware/clarity/discussions/6665#discussion-3865446

## How to reproduce
See screencast for the steps to reproduce the behavior:
https://user-images.githubusercontent.com/311619/141486153-2c73a13e-02a7-44a9-9493-cd4f854f75ed.mp4

### Describe the bug
Imagine the following quite common scenario:

1. User types in/selects a value from the suggestion-list of the autocomplete-combobox.
2. This value is valid and eg. saved until a change of the value is required.
3. User now clicks on the caret-down to open the suggestion-list, as he/she wants to select another value.
In the [screencast example](https://user-images.githubusercontent.com/311619/141486153-2c73a13e-02a7-44a9-9493-cd4f854f75ed.mp4) the user had previously selected "Alaska" and wants to change it to "Arizona", which is not easy possible.

### Current behaviour
- when the user clicks on the caret-down to open the suggestion-list, as he/she wants to select another value, the suggestion-list only offers the current value.

- Some of our users where confused if there aren't no other valid options to select from, and

- other understood that they need to delete the current value and search for a new one but felt that this behavior is cumbersome and unintuitive.

### Expected behavior
Therefore the behaviour should change as soon as a value has been selected = the combobox field is populated:

- then a click to open the suggestion-list should show all options, making the selection of another valid option via mouse easy.
- when the text of the input field is altered, the suggestion list reflects the search - behaviour and adapts the search results in the suggestion-list as currently implemented.
additionally a "x" at the end of the input field could help to easily clear the contents of the combobox. See common behaviour on search fields: 
<img width="290" alt="141487814-c261d15e-faeb-4584-8397-6181fff74c67" src="https://user-images.githubusercontent.com/311619/201676928-39954db7-b78e-41c1-90a1-ea90dbfc98e9.png">

see also https://github.com/vmware/clarity/discussions/5641

**Comment by Jinnie** [comment]
This is in accordance W3C accessibility spec for combobox.
I personally agree it may be considered unintuitive, as this is a topic that arises over and over again, but the W3C aria examples for editable combobox, which we use as basis for our implementation, are featuring filtered dropdown content:
https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-autocomplete-list.html

The, close to combobox, but native, datalist html element behaves exactly the same way:
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_datalist

I'll leave this open for a couple of weeks more, to possibly get feedback from our A11Y team, but currently we do not intend to change this behavior.

CC @AmyLiNow

**Comment by Jinnie** [comment]
Because it does not make sense to re-log the same problem again, then give the same comments again, etc. I want to summarize where we're at and CC the teams/people that can take a decision or prioritize this.

- Our combobox has 2 modes - single selection and multi selection. This bug is about single selection only.
- In single selection, the component behaves much like the native datalist. It shares the input field for both filtering purposes and showing the current selection. This and this W3C spec were driving our decision to implement it as it is now:
https://www.w3.org/WAI/ARIA/apg/example-index/combobox/combobox-autocomplete-list.html
- The proposed change is to not apply the filter on the list in the dropdown when it gets initially expanded.
- A11Y team says we are free to change the implementation if we decide to. They do not bind us by the W3C example, as long as the solution is not breaking accessibility.

Why we don't want to change it:
- We have a W3C spec, which we prefer to adhere to.
- We believe the customers' expectations should be in sync with the native datalist behavior.
- It's largely a matter of preference. There is currently no research backed with data on the users preferences and expectations here.

What can be done:
- Long term, we may split the implementations of single select and multi select comboboxes, as they are quite different in nature. Also, the multi select combobox might benefit from some architectural changes, as its current design is plagued with unresolvable a11y challenges. A fine-detail configuration may be added to enable/disable the requested feature.
- Short term, we may add such configuration option to the current design.
- We may first want to do a research to verify this with our users, what are their expectations and the importance weight they give to this issue. With our current resources this would be valuable.

CC @pdaigle @colinreedmiller

**Comment by d-m-s** [comment]
**Thanks @Jinnie for the sumup, and for pursuing the issue!**

**Proposal**
We [@porscheinformatik](https://github.com/porscheinformatik) can implement the **Expected behavior** (as described on top of this page) for the single selection only, and contribute the code to clarity. 

**Why we want to change it**
As mentioned, we do have lot of users feedback from different software products that the current behaviour of the component is cumbersome and hindering the editing of data. But if desired, we can also implement it using a parameter, so the new behaviour only occurs if the parameter is activated.

**Suggested next steps:**
- Please align the proposal with your team/stakeholders, and give us feedback if a contribution is desired and you're willing to overtake a pull-request to clr-repo.

**Comment by kevinbuhmann** [comment]
>But if desired, we can also implement it using a parameter, so the new behaviour only occurs if the parameter is activated.

My opinion would be not to make this behavior configurable.

**Comment by d-m-s** [comment]
Hi guys, 

as suggested above, we could move on and implement the behavior, as soon as we have some feedback from your side!

- I agree with @kevinbuhmann that one default behavior would be easier and primarily aim for that. 
Still if anyone has second thoughts, we can still go with an optional solution (via directive/flag/...)
- We can offer to make a short developer-sync-meeting on how to implement it (ms teams, slack, ...) if you guys like to.

@pdaigle @kevinbuhmann @Jinnie can you please pick this issue up, thanks in advance.

**Comment by d-m-s** [comment]
Thanks @Jinnie for setting up the demo!

> If I understand correctly, what we want is the 3rd option, the "Advanced (manually controlled) list" to be the default Combobox behavior (or configurable?).

Yes, exactly!

> We can do it, (or you can contribute it), but I want this demo validated by A11Y, Design and PM before we can work on it.

Please drive/push these validations, and give us feedback! (CC @Wallabeng, @poifir)

> I also noticed there is some not so pleasant scrolling behavior if the selection is far from the beginning of the list

A scroll-to the currently selected value could fix that, quit similar as the native "select"-box component.
(The selected value is shown as last item in the select-box. See [screencast](https://user-images.githubusercontent.com/311619/214919062-87355138-b179-426c-aa34-43283e63a523.mp4), first keyboard-nav, second mouse-nav used)

**Comment by Jinnie** [comment]
Pending ship today in the v15 RC.
You can see preview in the associated PR
